### PR TITLE
complete merge of async code

### PR DESCRIPTION
**Pull request by edhartnett**
Removed the non-async version of the code from the build system and the code base.

This will make it easier for contributors to submit modifications.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,6 @@ mark_as_advanced(VERSION_MAJOR VERSION_MINOR VERSION_PATCH)
 #===== Library Options =====
 option (PIO_ENABLE_FORTRAN   "Enable the Fortran library builds"            ON)
 option (PIO_ENABLE_TIMING    "Enable the use of the GPTL timing library"    ON)
-option (PIO_ENABLE_ASYNC     "Enable the use of asychronus IO operations"   OFF)
```
Comment (jedwards4b) [inline CMakeLists.txt]: async code is now always enabled?

File: src/clib/pio_darray.c [hunk 2]
```diff
@@ -226,8 +249,13 @@ int pio_write_darray_nc(file_desc_t *file, io_desc_t *iodesc, const int vid,
 #endif /* _NETCDF4 */
 	    case PIO_IOTYPE_NETCDF:
 	    {
+		/* Find the type size (again?) */
 		mpierr = MPI_Type_size(iodesc->basetype, &dsize);
+		
 		size_t tstart[ndims], tcount[ndims];
+
+		/* The IO master task does all the data writes, but
```
Comment (jedwards4b) [inline src/clib/pio_darray.c]: When using the serial netcdf PIO acts like a funnel.   Data is moved from compute tasks to IO tasks then from IO tasks to IO task 0 for write.  For read data the opposite happens, data is read on IO task 0 distributed to all io tasks and then to compute tasks.   We could optimize these data paths but serial netcdf is not the primary mode for PIO.   Can you change this comment?

File: src/clib/pio_darray.c [hunk 3]
```diff
@@ -372,50 +397,65 @@ int pio_write_darray_nc(file_desc_t *file, io_desc_t *iodesc, const int vid,
...
+ * @return 0 for success, error code otherwise.
+ * @ingroup PIO_write_darray
  */
-int pio_write_darray_multi_nc(file_desc_t *file, const int nvars, const int vid[],
-			      const int iodesc_ndims, MPI_Datatype basetype, const PIO_Offset gsize[],
+int pio_write_darray_multi_nc(file_desc_t *file, const int nvars, const int *vid,
+			      const int iodesc_ndims, MPI_Datatype basetype, const PIO_Offset *gsize,
 			      const int maxregions, io_region *firstregion, const PIO_Offset llen,
 			      const int maxiobuflen, const int num_aiotasks,
-			      void *IOBUF, const int frame[])
```
Comment (jedwards4b) [inline src/clib/pio_darray.c]: Is there a difference between const int frame[] and const int *frame?  Personal preference?

File: CMakeLists.txt (hunk 1 above)
Comment (edhartnett) [inline CMakeLists.txt]: Correct.

File: src/clib/pio_darray.c (hunk 3 above)
Comment (edhartnett) [inline src/clib/pio_darray.c]: Strong C convention.

File: src/clib/pio_darray.c (hunk 2 above)
Comment (edhartnett) [inline src/clib/pio_darray.c]: Done, thanks. Any updates to the comments are always welcome.

**Comment by jedwards4b** [comment]
Merging this PR with the understanding that pio_iosystem_tests2 will be addressed next.

**Pull request merged**